.github/workflows/ci.yaml

```diff
@@ -70,13 +48,27 @@ jobs:
         with:
           command: fmt
           args: --all -- --check
-
-  clippy:
-    runs-on: ubuntu-latest
+      
+      - name: Run clippy
+        run: make clippy
+  test:
+    name: Test
+    runs-on: macos-latest
+    env:
+      CARGO_TERM_COLOR: always
     steps:
       - uses: actions/checkout@v3
       - uses: actions-rs/toolchain@v1
         with:
-            toolchain: stable
-            components: clippy
-      - run: make clippy
+          toolchain: stable
+          components: clippy
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Generate code coverage
+        run: cargo llvm-cov --all-features --workspace --lcov --output-path lcov.info
```

Comment:
This step runs the tests and also generates the code coverage, maybe we can add that to the name, like `Run tests and generate code coverage`